libtransmission/peer-mgr-wishlist.cc

```diff
@@ -495,6 +497,30 @@ std::vector<tr_block_span_t> Wishlist::Impl::next(
             continue;
         }
 
+        // In sequential download mode, try to hotswap blocks that are already requested by slower peers
+        if (mediator_.is_sequential_download())
+        {
+            for (auto block = candidate.block_span.begin; block < candidate.block_span.end; ++block)
```

Comment:
@tearfur , I rebased my changes onto a 2a3a8ea3643d1ecb654f6ff38fa03b7e6683a8b1 where you introduced `unrequested`. It would be better to iterate over `requested` blocks instead of iterating over all blocks in my current changes.